solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -813,6 +818,11 @@ public NamedList<Object> request(SolrRequest<?> request, String collection)
     } else if (collection == null) {
       collection = defaultCollection;
     }
+    // TODO if we like this, then "defaultCollection" should be renamed
```

Comment:
[0] IMO this would be a bit inflexible, at least for a ticket that only set out to remove a setter.